ui/raidboss/emulator/data/PopupTextAnalysis.js

```diff
@@ -1,20 +1,68 @@
 import EmulatorCommon from '../EmulatorCommon';
 import StubbedPopupText from '../overrides/StubbedPopupText';
 
-export default class PopupTextAnalysis extends StubbedPopupText {
-  OnTriggerInternal(trigger, matches, currentTime) {
-    this.currentTriggerStatus = {
-      initialData: EmulatorCommon.cloneData(this.data),
-      condition: undefined,
-      response: false,
-      result: false,
-      delay: false,
-      suppressed: false,
-      executed: false,
-      promise: undefined,
+class Resolver {
+  constructor() {
+    this.promise = null;
+    this.run = null;
+    this.delayTo = null;
+  }
+  async IsResolved(log) {
+    if (this.delayTo) {
+      if (this.delayTo < log.timestamp) {
+        this.delayTo = null;
+        this.delay();
```

Comment:
Can this be called `this.delayResolver()` like it was before?

Reply:
Just to clarify, the three properties should be:
```js
this.delayTo;
this.delayPromise;
this.delayResolver;
```

Correct?

Reply:
I meannnnn if you're going to ask, I'd bikeshed and say `delayTo` => `delayUntil` for grammar reasons, but the other two are good.  Mostly I just think `this.delay()` operates differently than the other members and so it made it more readable to name it differently.

Reply:
Yeah, that makes sense. Changes made for next commit.